api/src/main/java/ai/djl/util/Ec2Utils.java

```diff
@@ -37,6 +44,26 @@ public final class Ec2Utils {
 
     private Ec2Utils() {}
 
+    /**
+     * Returns the SageMaker endpoint metadata for specified key.
+     *
+     * @param key the key to retrieve
+     * @return the SageMaker endpoint metadata for specified key
+     */
+    public static String readEndpointMetadata(String key) {
```

Comment:
We need update `readMetada()` function. readMetada() should be support for SM as well.

We need to get instance type with readMetada() function

Reply:
OK.